lib/sidekiq/web/application.rb

```diff
@@ -291,6 +294,30 @@ def call(env)
       resp
     end
 
+    def get_csp_header(scheme)
+      @csp_headers ||= {}
+      @csp_headers[scheme] ||= begin
+        return nil unless %w[https http].include?(scheme)
```

Comment:
A nil value will not memoize.  I'd suggest you use "" and change your guard to `if csp_header != ""`